src/app/fdctl/run/tiles/fd_store_int.c

```diff
@@ -0,0 +1,604 @@
+      if( FD_LIKELY( block ) ) {
+        block->flags = fd_uchar_set_bit( block->flags, FD_BLOCK_FLAG_PROCESSED );
+      }
+    } else {
+      return;
+    }
+  }
+  
+  ulong repair_req_cnt = 0;
+  switch( store_slot_prepare_mode ) {
+    case FD_STORE_SLOT_PREPARE_CONTINUE: {
+      if( slot > 64UL ) {
+        ctx->store->smr = fd_ulong_max( ctx->store->smr, slot - 64UL );
+        fd_pending_slots_set_lo_wmark( ctx->store->pending_slots, ctx->store->smr );
+      }
+      ctx->store->now = fd_log_wallclock();
+      break;
+    }
+    case FD_STORE_SLOT_PREPARE_NEED_PARENT_EXEC: {
+      break;
+    }
+    case FD_STORE_SLOT_PREPARE_NEED_REPAIR: {
+      repair_req_cnt = fd_store_slot_repair( ctx->store, slot, repair_reqs, MAX_REPAIR_REQS );
+      break;
+    }
+    case FD_STORE_SLOT_PREPARE_NEED_ORPHAN: {
+      fd_repair_request_t * repair_req = &repair_reqs[0];
+      repair_req->slot = slot;
+      repair_req->shred_index = UINT_MAX;
+      repair_req->type = FD_REPAIR_REQ_TYPE_NEED_ORPHAN;
+      repair_req_cnt = 1;
+      break;
+    }
+    case FD_STORE_SLOT_PREPARE_ALREADY_EXECUTED: {
+      return;
+    }
+    default: {
+      FD_LOG_ERR(( "unrecognized store slot prepare mode" ));
+      return;
+    }
+  }
+  
+  if( store_slot_prepare_mode == FD_STORE_SLOT_PREPARE_CONTINUE ) {
+
+    ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+    ulong replay_sig = fd_disco_replay_sig( slot, REPLAY_FLAG_FINALIZE_BLOCK );
+    uchar * out_buf = fd_chunk_to_laddr( ctx->replay_out_mem, ctx->replay_out_chunk );
+
+    fd_block_t * block = fd_blockstore_block_query( ctx->blockstore, slot );
+    if( block == NULL ) {
+      FD_LOG_ERR(( "could not find block" ));
+    }
+
+    fd_slot_meta_t * slot_meta = fd_blockstore_slot_meta_query( ctx->blockstore, slot );
+    if( slot_meta == NULL ) {
+      FD_LOG_ERR(( "could not find slot meta" ));
+    }
+
+    fd_hash_t const * block_hash = fd_blockstore_block_hash_query( ctx->blockstore, slot );
+    if( block_hash == NULL ) {
+      FD_LOG_ERR(( "could not find slot meta" ));
+    }
+
+    FD_STORE( ulong, out_buf, slot_meta->parent_slot );
+    out_buf += sizeof(ulong);
+
+    memcpy( out_buf, block_hash->uc, sizeof(fd_hash_t) );
+    out_buf += sizeof(fd_hash_t); 
+
+    uchar * block_data = fd_blockstore_block_data_laddr( ctx->blockstore, block );
+
+    FD_SCRATCH_SCOPE_BEGIN {
+      fd_block_info_t block_info;
+      fd_runtime_block_prepare( block_data, block->data_sz, fd_scratch_virtual(), &block_info );
+    
+      FD_LOG_NOTICE(( "block prepared - slot: %lu", slot ));
+
+      fd_txn_p_t * txns = fd_type_pun( out_buf );
+      ulong txn_cnt = fd_runtime_block_collect_txns( &block_info, txns );
+
+      ulong out_sz = sizeof(ulong) + sizeof(fd_hash_t) + ( txn_cnt * sizeof(fd_txn_p_t) );
+      fd_mcache_publish( ctx->replay_out_mcache, ctx->replay_out_depth, ctx->replay_out_seq, replay_sig, ctx->replay_out_chunk, txn_cnt, 0UL, tsorig, tspub );
```

Comment:
Why aren't you using the primary out for these? It seems like they need to be flow controlled and reliable.